bundles/org.openhab.binding.tplinkrouter/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<thing:thing-descriptions bindingId="tplinkrouter"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xmlns:thing="https://openhab.org/schemas/thing-description/v1.0.0"
+	xsi:schemaLocation="https://openhab.org/schemas/thing-description/v1.0.0 https://openhab.org/schemas/thing-description-1.0.0.xsd">
+
+	<!-- Sample Thing Type -->
+	<thing-type id="TD-W9970">
+		<!-- Comment this in for Things which need a Bridge to function.
+			<supported-bridge-type-refs>
+			<bridge-type-ref id="tplinkrouterBridge" />
+			</supported-bridge-type-refs>
+		-->
+
+		<label>TpLinkRouter Binding Thing</label>
+		<description>Sample thing for TpLinkRouter Binding</description>
+
+		<channel-groups>
+			<channel-group id="wifi" typeId="wifiGroupType"/>
+		</channel-groups>
+
+		<config-description>
+			<parameter name="hostname" type="text" required="true">
+				<context>network-address</context>
+				<label>Hostname</label>
+				<description>Hostname or IP address of the device</description>
+			</parameter>
+			<parameter name="username" type="text" required="true">
+				<context>username</context>
```

Comment:
To be checked but I believe this is not a valid context ?

Reply:
Is a parameter without context allowed?

Reply:
Yes, it is optional. 
You can find the table of valid contexts here: https://www.openhab.org/docs/developer/bindings/config-xml.html#supported-contexts